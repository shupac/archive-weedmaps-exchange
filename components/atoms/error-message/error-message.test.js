import React from 'react';
import renderer from 'react-test-renderer';
import ErrorMessage from './';

describe('ErrorMessage', () => {
  it('should render error state 404', () => {
    const tree = renderer.create(<ErrorMessage statusCode={404} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render error state 500', () => {
    const tree = renderer.create(<ErrorMessage statusCode={500} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render error state undefined', () => {
    const tree = renderer.create(<ErrorMessage statusCode={1000} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
