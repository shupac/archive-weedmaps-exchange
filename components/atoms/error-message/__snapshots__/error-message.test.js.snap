// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage should render error state 404 1`] = `
.c0 {
  padding: 0 15px;
}

.c0 h1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 2.625rem;
  margin: 0;
  padding-top: 64px;
  padding-bottom: 24px;
}

.c0 div {
  font-weight: 500;
  font-size: 1.25rem;
  text-align: center;
}

<div
  className="c0"
>
  <h1>
    404
  </h1>
  <div>
    We're sorry. The page you're looking for does not exist.
  </div>
</div>
`;

exports[`ErrorMessage should render error state 500 1`] = `
.c0 {
  padding: 0 15px;
}

.c0 h1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 2.625rem;
  margin: 0;
  padding-top: 64px;
  padding-bottom: 24px;
}

.c0 div {
  font-weight: 500;
  font-size: 1.25rem;
  text-align: center;
}

<div
  className="c0"
>
  <h1>
    500
  </h1>
  <div>
    Internal server error.
  </div>
</div>
`;

exports[`ErrorMessage should render error state undefined 1`] = `
.c0 {
  padding: 0 15px;
}

.c0 h1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 2.625rem;
  margin: 0;
  padding-top: 64px;
  padding-bottom: 24px;
}

.c0 div {
  font-weight: 500;
  font-size: 1.25rem;
  text-align: center;
}

<div
  className="c0"
>
  <h1>
    1000
  </h1>
  <div>
    Something went wrong. Please refresh and try again.
  </div>
</div>
`;
